client/extensions/woocommerce/app/products/product-form-delivery-details-card.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormDimensionsInput from '../../components/form-dimensions-input';
+import FormFieldSet from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+
+const ProductFormDeliveryDetailsCard = ( { product, editProduct, translate } ) => {
+	const setDimension = ( e ) => {
+		const dimensions = { ...product.dimensions, [ e.target.name ]: e.target.value };
+		editProduct( product, { dimensions } );
+	};
+
+	const { dimensions } = product;
+
+	// TODO Pull in dimensions unit from settings API.
+	return (
+		<Card className="products__product-form-delivery-details">
+			<FormFieldSet>
+				<FormLabel>{ translate( 'Dimensions' ) }</FormLabel>
+				<FormDimensionsInput
+					unit="in"
```

Comment:
The unit will actually come from the API once we have this hooked up, but I can wrap this in a translate in the meantime.